Tests/OpenCombineTests/PublisherTests/FutureTests.swift

```diff
@@ -0,0 +1,140 @@
+    private typealias Sut = Future<Int, TestingError>
+
+    func testFutureSuccess() {
+        var promise: Sut.Promise?
+
+        let future = Sut { promise = $0 }
+
+        let subscriber = TrackingSubscriber(receiveSubscription: { subscription in
+            subscription.request(.unlimited)
+        })
+
+        future.subscribe(subscriber)
+        promise?(.success(42))
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future"),
+            .value(42),
+            .completion(.finished)
+        ])
+    }
+
+    func testFutureFailure() {
+        var promise: Sut.Promise?
+
+        let future = Sut { promise = $0 }
+
+        let subscriber = TrackingSubscriber(
+            receiveSubscription: { subscription in
+                subscription.request(.unlimited)
+            }, receiveValue: { _ in
+                XCTFail("no value should be returned")
+                return .none
+            }
+        )
+
+        future.subscribe(subscriber)
+
+        let error = TestingError(description: "\(#function)")
+        promise?(.failure(error))
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future"),
+            .completion(.failure(error))
+        ])
+    }
+
+    func testResolvingMultipleTimes() {
+        var promise: Sut.Promise?
+
+        let future = Sut { promise = $0 }
+
+        let subscriber = TrackingSubscriber(receiveSubscription: { subscription in
+            subscription.request(.unlimited)
+        })
+
+        future.subscribe(subscriber)
+
+        promise?(.success(42))
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future"),
+            .value(42),
+            .completion(.finished)
+        ])
+
+        promise?(.success(41))
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future"),
+            .value(42),
+            .completion(.finished)
+        ])
+    }
+
+    func testCancellation() {
+        var promise: Sut.Promise?
+
+        let future = Sut { promise = $0 }
+
+        let subscriber = TrackingSubscriber(receiveSubscription: { subscription in
+            subscription.request(.unlimited)
+        })
+        future.subscribe(subscriber)
+
+        subscriber.subscriptions.forEach { $0.cancel() }
+
+        promise?(.success(42))
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future")
+        ])
+    }
+
+    func testSubscribeAfterResolution() {
+        var promise: Sut.Promise?
+
+        let future = Sut { promise = $0 }
+        promise?(.success(42))
+
+        let subscriber = TrackingSubscriber(receiveSubscription: { subscription in
+            subscription.request(.unlimited)
+        })
+
+        future.subscribe(subscriber)
+
+        XCTAssertEqual(subscriber.history, [
+            .subscription("Future"),
+            .value(42),
+            .completion(.finished)
+        ])
+    }
+    
+    func testNoDemand() {
```

Comment:
We already have a number of tests that are called `testCrashesOnZeroDemand`.
```suggestion
    func testCrashesOnZeroDemand() {
```